napari/_tests/test_viewer.py

```diff
@@ -376,3 +376,16 @@ def test_reset_non_empty(make_napari_viewer):
     viewer = make_napari_viewer()
     viewer.add_points([(0, 1), (2, 3)])
     viewer.reset()
+
+
+def test_zero_scale_layer(make_napari_viewer):
```

Comment:
just an fyi: tests using `make_napari_viewer` require the _full_ Qt/GUI-compatible stack, and run a lot slower.  In general, if possible, it's nice to test things on a bit lower level (unless the issue is specifically something to do with the full gui integration).

So, in this case, you could accomplish the same test coverage without using `make_napari_viewer` just by directly instantiating an Image layer:
```python
def test_zero_scale_layer():
    with pytest.raises(ValueError, match='scale values of 0'):
        Image(np.random.rand(64, 64), scale=(0, 1))
``` 

(note also the `pytest.raises` syntax which allows you to both assert that a specific exception was raised, and that it matches a certain string... and see other comment about the `ValueError`)

Then this test can be moved into `napari/layers/_tests` somewhere, and it will be run in our headless tests (not just those that require the full GUI setup)